test/Field.test.tsx

```diff
@@ -1,545 +1,352 @@
-      let injected: any; /** FieldProps ;) */
-      const validate = jest.fn(noop);
-      ReactDOM.render(
-        <TestForm
-          validateOnChange={false}
-          render={(_formikProps: FormikProps<TestFormValues>) => (
-            <Field name="name" validate={validate}>
-              {({ field }: any) => (injected = field) && null}
-            </Field>
-          )}
-        />,
-        node
-      );
-      const { onChange } = injected;
-      onChange({ target: { name: 'name', value: 'hello' } });
-      expect(validate).not.toHaveBeenCalled();
-    });
-
-    it('calls validate during onBlur if present', () => {
-      const node = document.createElement('div');
-      let injected: any; /** FieldProps ;) */
-      const validate = jest.fn(noop);
-      ReactDOM.render(
-        <TestForm
-          validateOnBlur={true}
-          render={(_formikProps: FormikProps<TestFormValues>) => (
-            <Field name="name" validate={validate}>
-              {({ field }: any) => (injected = field) && null}
-            </Field>
-          )}
-        />,
-        node
-      );
-      const { onBlur } = injected;
-      onBlur({ target: { name: 'name' } });
-      expect(validate).toHaveBeenCalled();
-    });
-
-    it('does NOT call validate during onBlur if validateOnBlur is set to false', () => {
-      const handleBlur = jest.fn(noop);
-      const setFieldError = jest.fn(noop);
-      const validate = jest.fn(noop);
-      const tree = makeFieldTree({
-        name: 'name',
-        validate,
-        formik: {
-          registerField: noop,
-          unregisterField: noop,
-          handleBlur,
-          setFieldError,
-          validateOnBlur: false,
-        },
-      });
-      tree.find('input').simulate('blur', {
-        persist: noop,
-        target: {
-          name: 'name',
-          value: 'ian',
-        },
-      });
-      expect(handleBlur).toHaveBeenCalled();
-      expect(validate).not.toHaveBeenCalled();
-    });
+const createRenderField = (
+  FieldComponent: React.ComponentClass<FieldConfig>
+) => (
+  props: Partial<FieldConfig> = {},
+  formProps?: Partial<FormikConfig<Values>>
+) => {
+  let injected: FieldProps | FastFieldProps;
+
+  if (!props.children && !props.render && !props.component) {
+    props.children = (fieldProps: FieldProps | FastFieldProps) =>
+      (injected = fieldProps) && null;
+  }
+
+  return {
+    getProps() {
+      return injected;
+    },
+    ...renderForm(<FieldComponent name="name" {...props} />, formProps),
+  };
+};
+
+const renderField = createRenderField(Field);
+const renderFastField = createRenderField(FastField);
+
+function cases(
+  title: string,
+  tester: (arg: typeof renderField | typeof renderFastField) => void
+) {
+  describe(title, () => {
+    it('<Field />', async () => Promise.resolve(tester(renderField)));
+    it('<FastField />', async () => Promise.resolve(tester(renderFastField)));
+  });
+}
```

Comment:
Cool

Reply:
TIL!

Reply:
Yeah, I was thinking if we could use `Promise.resolve` trick in Formik instead of `isPromise` check.